ui/src/components/Canvas.tsx

```diff
@@ -248,158 +289,186 @@ function ResultBlock({ pod, id }) {
     </Box>
   );
 }
-
 const CodeNode = memo<Props>(({ data, id, isConnectable }) => {
   const store = useContext(RepoContext);
   if (!store) throw new Error("Missing BearContext.Provider in the tree");
   // const pod = useStore(store, (state) => state.pods[id]);
   const wsRun = useStore(store, (state) => state.wsRun);
   const ref = useRef(null);
-  const [target, setTarget] = React.useState<any>(null);
-  const [frame] = React.useState({
+  const [target, setTarget] = useState<any>(null);
+  const [frame] = useState({
     translate: [0, 0],
   });
+  const [isEditorBlur, setIsEditorBlur] = useState(true);
   // right, bottom
-  const [layout, setLayout] = useState("right");
+  const [layout, setLayout] = useState("bottom");
   const { setNodes } = useReactFlow();
   // const selected = useStore(store, (state) => state.selected);
   const setSelected = useStore(store, (state) => state.setSelected);
+  const nodesMap = useStore(store, (state) => state.ydoc.getMap<Node>("pods"));
+  const onResize = useCallback((e, data) => {
+    const { size } = data;
+    const node = nodesMap.get(id);
+    if (node) {
+      node.style = { ...node.style, width: size.width };
+      nodesMap.set(id, node);
+      updatePod({
+        id,
+        data: {
+          width: size.width,
+          height: pod.height,
+        },
+      });
+    }
+  }, []);
+  const onLayout = useCallback(({ height }) => {
+    const node = nodesMap.get(id);
+    if (node) {
+      node.style = { ...node.style, height };
+      nodesMap.set(id, node);
+    }
+  }, []);
```

Comment:
These were removed in #61. So there are still merging problems. There might be other places, it's tedious to check. If it is not easy to merge, maybe let's create a clean PR from the current `main` branch?

Reply:
this is for width adjustment, it's different from removed one